docs/src/extend/code-path-analysis.md

```diff
@@ -271,75 +403,101 @@ function isCbCalled(info) {
-            });
-        },
-        "onCodePathEnd": function(codePath, node) {
-            funcInfoStack.pop();
-
-            // Checks `cb` was called in every paths.
-            var cbCalled = codePath.finalSegments.every(function(segment) {
-                var info = segmentInfoMap[segment.id];
-                return info.cbCalled;
-            });
-
-            if (!cbCalled) {
-                context.report({
-                    message: "`cb` should be called in every path.",
-                    node: node
+module.exports = {
+    meta: {
+        // ...
+    },
+    create(context) {
+
+        let funcInfo;
+        const funcInfoStack = [];
+        const segmentInfoMap = Object.create(null);
+
+        return {
+            // Checks `cb`.
+            onCodePathStart(codePath, node) {
+                funcInfoStack.push(funcInfo);
+
+                funcInfo = {
+                    codePath: codePath,
+                    hasCb: hasCb(node, context),
+                    currentSegments: new Set()
+                };
+            },
+
+            onCodePathEnd(codePath, node) {
+                funcInfo = funcInfoStack.pop();
+
+                // Checks `cb` was called in every paths.
+                const cbCalled = codePath.finalSegments.every(function(segment) {
+                    const info = segmentInfoMap[segment.id];
+                    return info.cbCalled;
                 });
-            }
-        },
 
-        // Manages state of code paths.
-        "onCodePathSegmentStart": function(segment) {
-            var funcInfo = funcInfoStack[funcInfoStack.length - 1];
-
-            // Ignores if `cb` doesn't exist.
-            if (!funcInfo.hasCb) {
-                return;
+                if (!cbCalled) {
+                    context.report({
+                        message: "`cb` should be called in every path.",
+                        node: node
+                    });
+                }
+            },
+
+            // Manages state of code paths and tracks traversed segments
+            onCodePathSegmentStart(segment) {
+
+                funcInfo.currentSegments.add(segment);
+
+                // Ignores if `cb` doesn't exist.
+                if (!funcInfo.hasCb) {
+                    return;
+                }
+
+                // Initialize state of this path.
+                const info = segmentInfoMap[segment.id] = {
+                    cbCalled: false
+                };
+
+                // If there are the previous paths, merges state.
+                // Checks `cb` was called in every previous path.
+                if (segment.prevSegments.length > 0) {
+                    info.cbCalled = segment.prevSegments.every(isCbCalled);
+                }
+            },
+
+            // Tracks unreachable segment traversal
+            onUnreachableCodePathSegmentStart(segment) {
+                funcInfo.currentSegments.add(segment);
             }
```

Comment:
```suggestion
            },
```

Just to fix syntax errors in this rule example.

Otherwise, the example doesn't work well, it crashes when I try it. However, the previous version didn't work well either, it crashes in the same way, so I think fixing it is out of scope for this PR. I'll fix it in a follow-up.